src/hooks/useSubmitButtonVisibility.ts

```diff
@@ -0,0 +1,73 @@
+import {useEffect, useRef, useState} from 'react';
+import {Dimensions} from 'react-native';
+import getPlatform from '@libs/getPlatform';
+import CONST from '@src/CONST';
+import useSafeAreaInsets from './useSafeAreaInsets';
+import useThemeStyles from './useThemeStyles';
+import useWindowDimensions from './useWindowDimensions';
+
+// Useful when there's a need to hide the submit button, from FormProvider,
+// to let form content fill the page when virtual keyboard is shown
+function useSubmitButtonVisibility() {
+    const styles = useThemeStyles();
+    const {windowHeight, isSmallScreenWidth} = useWindowDimensions();
+    const [isSubmitButtonVisible, setIsSubmitButtonVisible] = useState(true);
+    const initialWindowHeightRef = useRef(windowHeight);
+    const isSmallScreenWidthRef = useRef(isSmallScreenWidth);
```

Comment:
I think I'm missing something, could you elaborate on why we need initial values (why do we use `useRef`?)

Reply:
@eVoloshchak useRef is used here to store the initial value of window height, so that it can be compared with the current window height value to determine if the virtual keyboard is shown or not. About isSmallScreenWidthRef I had to use a ref too because the callback I pass to the `Dimensions.addEventListener` cannot have dependencies, otherwise multiple event listeners would be set 😄